### PR TITLE
feat: add basic local chat search

### LINKED ISSUE #2771: implement search modal & local message search
Implement search that opens a modal, and allows searching the messages of that channel that are stored locally

https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?node-id=6042%3A1

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #2771

Adds a very basic search feature. It only searches the current channel, and only messages that were locally loaded, so it is very limited. Another limit is that it only searches the raw message, so if there is a mention, you can't search the name. This should all be improved once the search is moved to status-go. I purposefully made it simple on the Nim side, since it will move anyway.

This also includes a new BadgeContent component that is both used in the ActivityCenter and the Search. It is used to show this: 
image

It is not currently possible to search more than the current channel or to change the channel searched. It will be improved in another issue/PR

image

image

### CODE REVIEW

File: ui/app/AppLayouts/Chat/ChatColumn/TopBar.qml [hunk 1]
```diff
@@ -70,6 +70,22 @@ Item {
...
+            icon.name: "search"
+            iconColor: Style.current.contextMenuButtonForegroundColor
+            hoveredIconColor: Style.current.contextMenuButtonForegroundColor
+            highlightedBackgroundColor: Style.current.contextMenuButtonBackgroundHoverColor
+            anchors.verticalCenter: parent.verticalCenter
+
+            onClicked:openPopup(searchPopopComponent)
+
+            Component {
+                id: searchPopopComponent
```
Comment (borismelnik) [inline ui/app/AppLayouts/Chat/ChatColumn/TopBar.qml:84]: just small stupid thing - `PopOp`?

File: ui/app/AppLayouts/Chat/components/BadgeContent.qml [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        if (chatType === Constants.chatTypeCommunity) {
+            // TODO add a check for private community chats once it is created
+            return Constants.chatTypePubliccommunityComponent
+        }
+        return chatType
+    }
+
+    property string profileImage: realChatType === Constants.chatTypeOneToOne ? appMain.getProfileImage(chatId) || ""  : ""
+
+    id: wrapper
```
Comment (borismelnik) [inline ui/app/AppLayouts/Chat/components/BadgeContent.qml:24]: Will be great if you move id to first line

**Review by borismelnik — APPROVED** [review]

**Review by iurimatias — CHANGES_REQUESTED** [review]
Hey good job on this, I gived it a spin, and the search modal height does grow all over the place depending on the search results, could we give it a max height or even a set height once search results show up to avoid this?

**Comment by jrainville** [comment]
Ah I did give it a max height, but forgot to test it. I'll adjust it to make it work better and also I forgot to add a scroll 🤦 
I'll fix tomorrow morning.

**Comment by anastasiyaig** [comment]
@jrainville can we please fix the problem Iury mentioned , also a few more:

- [ ] missing scroll in search modal
- [ ] fix the modal height
- [ ] put a contact name in a 1x1 chat search indicator
Screenshot 2021-07-12 at 11 16 14
- [ ] custom picture is not being applied in search pictogram
Screenshot 2021-07-12 at 12 08 05
- [ ] search icon (magnifying glass) seems to be different than in figma
Screenshot 2021-07-12 at 12 19 48

- [ ] hash symbol is missing when referring a channel in community (assume from designs it should be there), also the community indicator (grey rainbow) probably should not be there (it is not shown on designs)

Screenshot 2021-07-12 at 12 43 11

**Comment by jrainville** [comment]
@iurimatias and @anastasiyaig thanks for the tests.
Your issues should be fixed now

**Comment by anastasiyaig** [comment]
@jrainville thank you for the hard work, looks amazing! is it also possible to make the search non-global, so when entering a search criteria it is not saved there after i close and open the search modal?

here is what @simonricoo suggests:

1. It defo should be cleared when you switch and search the another channel. So you dont bring searched content from "channel 2" to "channel 1" 
2. When searching the same channel Im fine both ways either its remembered or not


https://user-images.githubusercontent.com/82375995/125447553-4ab6b4df-d0e4-44d7-a0ca-191fc1d17ddc.mov

it could be handled in another issue if this is something that will break the whole feature :D Please let me know what u think!

**Comment by jrainville** [comment]
I can fix it here. It's simple enough. I'll do it later today

**Comment by jrainville** [comment]
@anastasiyaig done

**Comment by anastasiyaig** [comment]
great job, thank you @jrainville

**Pull request merged**